### PR TITLE
jak2: add back some pc port debugging tools and add `palace-scenes` to gsrc

### PR DESCRIPTION
**Pull request by Hat-Kid**
Ports some of the custom PC port debugging code from Jak 1, such as the entity debugger, Anim Tester X and part tester, to Jak 2 and gives the PC menu code the same treatment Jak 1 received recently in #2216.
Some adjustments have been made to make things work in Jak 2 and some stuff is still incomplete.
The majority of the default PC menu code has been left commented out in preparation for @ManDude's plan to port the rest of the Jak 1 `pckernel` features to Jak 2.

Also adds the code for `palace-scenes` to gsrc as that seems to have been missed.

### CODE REVIEW

File: goal_src/jak2/engine/debug/part-tester.gc [hunk 1]
```diff
@@ -13,7 +17,7 @@
 (declare-file (debug))
 
 (defpartgroup group-part-tester
-  :id 127
+  :id 96
```
Comment (ManDude) [inline goal_src/jak2/engine/debug/part-tester.gc:20]: Why was this changed?

File: goal_src/jak2/engine/game/main.gc [hunk 2]
```diff
@@ -1276,6 +1276,48 @@
...
+  `(let* ((heap ,heap) (pool-total (memory-total heap)))
+      (draw-memory-bar-generic ,buf
+              :remain (- pool-total (memory-used heap))
+              :total  pool-total
+              :name ,(if name name (symbol->string heap))
+              :idx ,idx
+              :color ,color)
+      )
+  )
+
```
Comment (ManDude) [inline goal_src/jak2/engine/game/main.gc:1320]: Pretty happy all of the new stuff I wrote still works fine even in a completely different game!

**Review by ManDude — CHANGES_REQUESTED** [review]
Aside from the thing in `part-tester` it all looks fine to me. I'd like to have this merged before going forward with the `pckernel` porting.

File: goal_src/jak2/engine/debug/part-tester.gc (hunk 1 above)
Comment (Hat-Kid) [reply goal_src/jak2/engine/debug/part-tester.gc:20]: Main reason was that the `let` on line 71 was commented out and replaced with a `let` that uses id 96 instead for whatever reason. I don't think it matters much which id is used since neither seem to be used by the game.

Comment (ManDude) [reply goal_src/jak2/engine/debug/part-tester.gc:20]: I think that was just them testing whatever 96 was at the time. IMO just keep it at 127, there's checks in place to make sure invalid part groups aren't tested anyway.

**Review by ManDude — APPROVED** [review]
